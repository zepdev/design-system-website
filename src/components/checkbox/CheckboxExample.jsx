import React, { useState } from "react"
import Checkbox from "@zlab-de/zel-react/Checkbox"

function CheckboxExample() {
  const [state, setState] = useState({
    isCheckedA: true,
    isCheckedB: true,
    isCheckedF: true,
  })

  const handleChange = name => event => {
    setState({ ...state, [name]: event.target.checked })
  }

  return (
    <form>
      <Checkbox
        label="A"
        isChecked={state.isCheckedA}
        handleChange={handleChange("isCheckedA")}
      />
      <Checkbox
        label="B"
        isChecked={state.isCheckedB}
        handleChange={handleChange("isCheckedB")}
      />
      <Checkbox
        label="C"
        isChecked={state.isCheckedC}
        handleChange={handleChange("isCheckedC")}
      />
    </form>
  )
}
export default CheckboxExample
