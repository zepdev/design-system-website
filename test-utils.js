import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
