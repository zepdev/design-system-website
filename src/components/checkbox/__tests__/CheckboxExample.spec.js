import React from "react"
import { render } from "../../../../test-utils"
import CheckboxExample from "../CheckboxExample"

describe("CheckboxExample", () => {
  it("renders correctly", () => {
    const component = render(<CheckboxExample />)
    expect(component).toMatchSnapshot()
  })
})
