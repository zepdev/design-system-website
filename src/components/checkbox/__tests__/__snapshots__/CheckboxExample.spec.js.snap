// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <input
          checked=""
          class="zep-checkbox"
          id="checkbox_A"
          type="checkbox"
        />
        <label
          class="zep-checkbox__label"
          for="checkbox_A"
        >
          A
        </label>
        <input
          checked=""
          class="zep-checkbox"
          id="checkbox_B"
          type="checkbox"
        />
        <label
          class="zep-checkbox__label"
          for="checkbox_B"
        >
          B
        </label>
        <input
          class="zep-checkbox"
          id="checkbox_C"
          type="checkbox"
        />
        <label
          class="zep-checkbox__label"
          for="checkbox_C"
        >
          C
        </label>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <input
        checked=""
        class="zep-checkbox"
        id="checkbox_A"
        type="checkbox"
      />
      <label
        class="zep-checkbox__label"
        for="checkbox_A"
      >
        A
      </label>
      <input
        checked=""
        class="zep-checkbox"
        id="checkbox_B"
        type="checkbox"
      />
      <label
        class="zep-checkbox__label"
        for="checkbox_B"
      >
        B
      </label>
      <input
        class="zep-checkbox"
        id="checkbox_C"
        type="checkbox"
      />
      <label
        class="zep-checkbox__label"
        for="checkbox_C"
      >
        C
      </label>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
